### ISSUE TITLE
Python 3.12 blocker in govcloud regions

### ISSUE THREAD

**Issue by david-saeger** [awslabs/LISA#92]
The Python 3.12 runtime is not currently available in GovCloud regions. Attempting to deploy on v3 the [xxx]-[xxx]-models-dev stack failed. The logical ID from CloudFormation of the error is `ModelsApidockerimagebuilderowlowlmodelsdevdockerimagebuilder295C9303` and the error message 
```
Resource handler returned message: "Access denied for runtime python3.12. Specify a different runtime and retry the request. (Service: Lambda, Status Code: 400, Request ID: 5b7c2626-7c1b-4d44-a48a-e7ae3ac403a6)" (RequestToken: b4dc9729-752e-1899-5caa-627ff04ede8b, HandlerErrorCode: InvalidRequest)
```

I see 3.12 was added as a dependency here https://github.com/awslabs/LISA/blob/0e824eb42dd779ab93c6976d2bcde94271c20f50/lib/models/docker-image-builder.ts#L115 and in a couple other places. Curious how deeply reliant v3 is on the latest python. Might have to fall back to v2 or come up with another plan.

**Comment by petermuller** [comment]
Good catch! As far as I know, we're not reliant on anything in the most recent python version- I know just from personal projects that 3.12 breaks some things that I normally use (although I can't remember what those are at the moment 😬 )

Prior to this set of changes, we've been relying on the python version in the config.yaml file, which i've had success with 3.9 and 3.10, and I haven't tried 3.11. I think 3.9 or 3.10 would be safe versions to try instead of 3.12 if GovCloud is lacking the most recent versions.

if 3.9 or 3.10 work without a problem, then we should make the updates to the props to add the Config object and then we can plug in `config.lambdaConfig.pythonRuntime` instead of the hardcoded version.

Just based on the schema.ts file, the max we supported so far is 3.11, so this one was just an oversight.

---

tl;dr, I expect 3.9 or 3.10 to work fine instead of 3.12

**Comment by petermuller** [comment]
Related review merged. We will now pull the Python version that is defined in config.yaml for the model deployment pieces. Closing this issue 👍
